### ISSUE TITLE
`import/order`: alphabetize `import type` above (or below) `import`

### ISSUE THREAD

**Issue by aaronadamsCA** [import-js/eslint-plugin-import#2339]
> For the version where types are mixed within groups, it would be nice if ^ was actually enforced. We have a bunch of
> 
> ```
> import type A from 'a';
> import A from 'a';
> import B from 'b';
> import type B from 'b';
> ```
> 
> and it would be great to have a more strict ordering if the library is going to start being type-import aware. For reference, [@typescript-eslint/consistent-type-imports](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-type-imports.md) extracts type imports and places them directly above their corresponding regular imports (though it has no on-going opinion after the initial extraction).
> 
> _Originally posted by @merrywhether in https://github.com/import-js/eslint-plugin-import/issues/2070#issuecomment-841732870_

This would be nice. The main reason we sort imports is to minimize merge conflicts, and I've noticed when sorting with this plugin that it has no opinion on the order between these two lines:

```ts
import { foo } from "./foo";
import type { Foo } from "./foo";
```

Both `simple-import-sort` and, as mentioned, `@typescript-eslint/consistent-type-imports` place `import type` (or `export type`) above `import` (or `export`) when the paths are identical; that seems like it could be a good precedent to follow. 

This might be as easy as concatenating `importKind` within the alphabetize comparator; I tried this with a local monkey patch and it seemed to do the trick.

**Comment by aaronadamsCA** [comment]
As an amusing side effect, this terrible monkey patch also appears to perfectly resolve #1881, since the alphabetization algorithm is slightly influenced by the appended text.

```js
  const comparator = alphabetizeOptions.caseInsensitive
    ? (a, b) => sorterFn(`${a.value.toLowerCase()}|${a.node.importKind}`, `${b.value.toLowerCase()}|${b.node.importKind}`)
    : (a, b) => sorterFn(`${a.value}|${a.node.importKind}`, `${b.value}|${b.node.importKind}`);
```

**Comment by ljharb** [comment]
@aaronadamsCA that type of comparison is what it used to do, and that caused bugs. That patch would break tests.

**Comment by ljharb** [comment]
I would expect type imports to always sort after value ones, since the types are far far less important and valuable.

I agree that this plugin should have an opinion about which order it sorts those in.

**Comment by aaronadamsCA** [comment]
Note that I filed #2398 as a first attempt at solving this.
